### PR TITLE
Fix TypeError when passing unicode args to agg gradient methods

### LINKED ISSUE #342: TypeError when calling linear_gradient
The `CtfEditor` in `ensemble` uses `GraphicsContext.linear_gradient`. The PR https://github.com/enthought/ensemble/pull/65 adds a `from __future__ import unicode_literals` import to the top of the module, which leads to a `TypeError` in the test output on CI.

The traceback is 
```
test_mouse_drag_alpha (ensemble.ctf.tests.test_editor.TestEditor) ... Traceback (most recent call last):
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/qt4/base_window.py", line 202, in paintEvent
    self.handler.paintEvent(event)
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/qt4/base_window.py", line 54, in paintEvent
    self._enable_window._paint(event)
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/abstract_window.py", line 468, in _paint
    self.component.draw(gc, view_bounds=(0, 0, size[0], size[1]))
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/component.py", line 427, in draw
    self._draw(gc, view_bounds, mode)
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/component.py", line 779, in _draw
    self._dispatch_draw(layer, gc, view_bounds, mode)
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/enable/container.py", line 257, in _dispatch_draw
    my_handler(gc, view_bounds, mode)
  File "/home/travis/build/enthought/ensemble/ensemble/ctf/editor.py", line 183, in _draw_container_mainlayer
    self._draw_colors(color_nodes, gc)
  File "/home/travis/build/enthought/ensemble/ensemble/ctf/editor.py", line 209, in _draw_colors
    gc.linear_gradient(0, 0, w, 0, grad_stops, 'pad', 'userSpaceOnUse')
  File "/home/travis/.edm/envs/ensemble-test-2.7-pyside/lib/python2.7/site-packages/kiva/agg/agg.py", line 1500, in linear_gradient
    return _agg.GraphicsContextArray_linear_gradient(self, *args)
TypeError: in method 'GraphicsContextArray_linear_gradient', argument 7 of type 'char const *'
ok
```

Note that the tests themselves passed. This traceback wasn't seen earlier when the `unicode_literals` import wasn't present.

### PR DESCRIPTION
**Pull request by jwiggins**
Fixes #342

### CODE REVIEW

**Comment by corranwebster** [comment]
LGTM, but not a SWIG expert so I won't catch anything subtle.

File: kiva/tests/test_agg_drawing.py [hunk 1]
```diff
@@ -8,6 +10,18 @@ class TestAggDrawing(DrawingImageTester, unittest.TestCase):
...
+    def test_unicode_gradient_args(self):
+        color_nodes = [(0.0, 1.0, 0.0, 0.0), (1.0, 0.0, 0.0, 0.0)]
+        with self.draw_and_check():
+            w, h = self.gc.width(), self.gc.height()
+            grad_stops = np.array([(x, r, g, b, 1.0)
+                                   for x, r, g, b in color_nodes])
+
+            self.gc.rect(0, 0, w, h)
+            self.gc.linear_gradient(0, 0, w, 0, grad_stops,
+                                    u'pad', b'userSpaceOnUse')
```
Comment (jwiggins) [inline kiva/tests/test_agg_drawing.py]: This should exercise the bug effectively.

File: kiva/agg/src/agg_typemaps.i [hunk 2]
```diff
@@ -337,3 +337,34 @@
...
+    return NULL;
+  }
+}
+%typemap(freearg) const char* gradient_arg
+{
+    if (utfstr$argnum)
+    {
+        Py_DECREF(utfstr$argnum);
+    }
+}
```
Comment (jwiggins) [inline kiva/agg/src/agg_typemaps.i]: @transfix Can you give your opinion here whether I'm crossing my t's and dotting my i's w.r.t. Python memory management?

Comment (jwiggins) [reply kiva/agg/src/agg_typemaps.i]: [PyUnicode_AsUTF8String](https://docs.python.org/2.7/c-api/unicode.html#c.PyUnicode_AsUTF8String) returns a new string reference. If my reading of the C-API documentation is correct, this means that it's this code's responsibility to decrement the reference .

Comment (transfix) [reply kiva/agg/src/agg_typemaps.i]: Checks out to me. One suggestion, if possible, is to change the c string args of linear_gradient to const std::string& and use std_string.i that comes with swig.

Comment (jwiggins) [reply kiva/agg/src/agg_typemaps.i]: Thanks for looking!

**Pull request merged**